garak/generators/function.py

```diff
@@ -82,14 +79,12 @@ def __init__(
         self.generator = getattr(gen_module, gen_function_name)
         import inspect
 
-        if "generations" in inspect.signature(self.generator).parameters:
+        if "name" in inspect.signature(self.generator).parameters:
             raise ValueError(
-                'Incompatible function signature: "generations" is incompatible with this Generator'
+                'Incompatible function signature: "name" is incompatible with this Generator'
```

Comment:
A conflict during inline creation is what this test is intent on warning about.

``` Python
my_args = { "name": "my_module#my_function", "arg1": "val1" }
Single(**my_args)
```

The above is a valid way to instantiate the generator and in this case `name` would be extracted an not be in `kwargs` when passed to `my_module#my_function` so if `my_module#my_function` requires a `name` parameter in it's signature this would fail when `_call_model()` executes.

If the function passed in via the `name` parameter of creation were to also contain name as a `kwargs` value an attempt to pass it to Single directly should except before even reaching this code.

``` Python
>>> def my_function(prompt: str, name: str, arg1: str):
...     pass
...
>>> my_args = { "name": "test", "arg1": "val1" }
>>> my_function("inline_prompt", name="fake", **my_args)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __main__.my_function() got multiple values for keyword argument 'name'
```

This offers the test *could* be removed as the user likely will see the error earlier than this check. I expect the `function` generator to  be the most commonly instantiated directly so having extra checks seemed reasonable.

Also in theory this conflict could be bypassed by providing the `kwargs` via config as shown on L44, as that would avoid passing `name` and `kwargs` at the same level to `__init__`.

If removed, the two cases below could be supported so there is a trade of for early runtime warning or support of a parameter in the called function that I *hope* actually has a low collision rate.

``` yaml
plugins:
  function:
    Single:
      name: "my_module#my_function"
      kwargs:
        name: "my_name_value"
        arg1: "val1"
```

``` Python
Single(config_root=_config)
```

or 

``` Python
my_args = { "function" { "kwargs": { "name": "my_name_value", "arg1": "val1" } } }
Single(name="my_module#my_function", config_root=my_args)
```